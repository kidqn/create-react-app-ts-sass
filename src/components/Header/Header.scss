h1 {
    color: red;
}
.Header {
    height: 80px;
    overflow: hidden;
    text-align: center;
    background-color: black;
    color: #fff;
    & > span{
        height: 100%;
        margin: auto;
        padding-top: 20px;
        display: inline-block;
        width: auto;
    }
}
.Header-title {
    font-weight: 900;
    text-transform: uppercase;
}
.Header-desc {
    margin-top: 4px;
    text-align: right;
    span{
        font-weight: 300;
        margin: 0 4px;
    }
    & span:nth-of-type(1){
        color: cornflowerblue;
    }
    & span:nth-of-type(2){
        color: red;
    }
    & span:nth-of-type(3){
        color: yellow;
    }
}