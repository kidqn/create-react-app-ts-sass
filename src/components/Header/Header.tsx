import * as React from 'react';
import './Header.scss';

class Header extends React.Component<any,any>{
    render (){
        return (
            <div className="Header">
                <span>
                    <div className="Header-title">Best statement collections</div>
                    <div className="Header-desc">
                        <span>Work</span><span>The</span><span>Color</span>
                    </div>
                </span>
            </div>
        )
    }
}
export default Header;