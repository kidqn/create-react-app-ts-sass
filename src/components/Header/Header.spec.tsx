import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import Header from './Header'

describe('Header component', () => {
    it('contains class name Header', () => {
        const component = shallow(<Header />);
        expect(component.props().className).toBe('Header');
    })
    it('render a correct snapshot', () => {
        const component =renderer.create(<Header />).toJSON();
        expect(component).toMatchSnapshot();
    })
})