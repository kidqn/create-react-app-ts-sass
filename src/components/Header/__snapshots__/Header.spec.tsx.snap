// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component render a correct snapshot 1`] = `
<div
  className="Header"
>
  <span>
    <div
      className="Header-title"
    >
      Best statement collections
    </div>
    <div
      className="Header-desc"
    >
      <span>
        Work
      </span>
      <span>
        The
      </span>
      <span>
        Color
      </span>
    </div>
  </span>
</div>
`;
